src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryTemplateParticipationResource.java

```diff
@@ -0,0 +1,190 @@
+package de.tum.in.www1.artemis.web.rest.repository;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.notFound;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.service.connectors.ContinuousIntegrationService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+import de.tum.in.www1.artemis.web.rest.FileMove;
+import de.tum.in.www1.artemis.web.rest.dto.RepositoryStatusDTO;
+
+/**
+ * Executes repository actions on repositories related to the participation id transmitted. Available to the owner of the participation, TAs/Instructors of the exercise and Admins.
+ */
+@RestController
+@RequestMapping({ "/api", "/api_basic" })
+@PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+public class RepositoryTemplateParticipationResource extends RepositoryResource {
+
+    private final ProgrammingExerciseParticipationService participationService;
+
+    public RepositoryTemplateParticipationResource(UserService userService, AuthorizationCheckService authCheckService, Optional<GitService> gitService,
+            Optional<ContinuousIntegrationService> continuousIntegrationService, RepositoryService repositoryService,
+            ProgrammingExerciseParticipationService programmingExerciseParticipationService) {
+        super(userService, authCheckService, gitService, continuousIntegrationService, repositoryService);
+        this.participationService = programmingExerciseParticipationService;
+    }
+
+    @Override
+    Repository getRepository(Long participationId) throws IOException, IllegalAccessException, InterruptedException {
+        Optional<TemplateProgrammingExerciseParticipation> participation = participationService.findTemplateParticipation(participationId);
```

Comment:
This is an example where we now have duplication:
Because we can't get all programming exercise participation types from the same repository, we now have to use different endpoints for differentiation.

Reply:
This can be resolved by implementing a generic ParticipationRepository and type casting when needed. Thanks @krusche